runtime/common/src/parachains.rs

```diff
@@ -668,6 +712,36 @@ decl_module! {
 
 			Ok(())
 		}
+
+		/// Transfer some tokens into a parachain and leave a message in the downward queue for it.
+		#[weight = 100_000]
+		pub fn transfer_to_parachain(
+			origin,
+			to: ParaId,
+			amount: Balance,
+			remark: Remark,
+		) {
+			let who = ensure_signed(origin)?;
+			let downward_queue_count = DownwardMessageQueue::<T>::decode_len(to).unwrap_or(0);
+			ensure!(downward_queue_count < MAX_DOWNWARD_QUEUE_COUNT, Error::<T>::DownwardMessageQueueFull);
+			T::ParachainCurrency::transfer_in(&who, to, amount, ExistenceRequirement::AllowDeath)?;
+			DownwardMessageQueue::<T>::append(to, DownwardMessage::TransferInto(who, amount, remark));
+		}
+
+		/// Send a XCMP message to the given parachain.
+		///
+		/// The origin must be another parachain.
+		#[weight = 100_000]
+		pub fn send_xcmp_message(
```

Comment:
So here is the ordering of events:
  1. Parachain queues upwards message
  2. Upwards message is processed at some point in the future
  3. If there is space on the downwards message queue, success. Otherwise, the message is dropped silently. 

Is it possible to re-queue the upwards message?

Reply:
Yeah I know, this is a problem. As explained above, this is all just for demonstrating stuff. Maybe the best would be to have a back pressure mechanism as for downward messages. Aka, the relay chain says "currently I will not accept any upward messages from you".

Reply:
Yeah, this is fine, let's fix it for the new parachains runtime and not here.